frontend/src/store/middlewares/socket/chat-socket.middleware.ts

```diff
@@ -0,0 +1,48 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { SocketEvent, SocketNamespace } from 'common/enums/enums';
+import {
+  AppDispatch,
+  ChatMessageGetAllItemResponseDto,
+} from 'common/types/types';
+import { io } from 'socket.io-client';
+import { chatsActions } from 'store/actions';
+
+type SocketMiddlewareParams = {
+  dispatch: AppDispatch;
+};
+
+const chatSocketInstance = io(SocketNamespace.CHAT);
```

Comment:
you need to create a socket service. you can create a method that will return an instance when you provide a name as an argument

Reply:
So the only purpose of the service will be to return an instance?